Src/BootCamp.Chapter/MsFactory.cs

```diff
@@ -1,15 +1,71 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Threading;
 using BootCamp.Chapter.Computer;
+using BootCamp.Chapter.Computer.Parts;
 
 namespace BootCamp.Chapter
 {
-    public class MsFactory
+    public class MsFactory : Factory
     {
         public DesktopComputer Assemble()
         {
-            return new DesktopComputer();
+            var pc = new DesktopComputer();
```

Comment:
If you compare computer assembling in Ms and Mac, what's the difference? The assemble is exact same, except that individual steps are different.

Reply:
This flow (assembling) can be implemented in abstract class.